### PR TITLE
FAB-16437 Remove common/mocks from v12 validator

### PR DESCRIPTION
**Pull request by mastersingh24**
#### Type of change

Improvement

#### Description

Mocks should be generated in the package in which they are used.
This is part of removing the use of common/mocks as it is a
bad pattern.

- stop using common/mocks in core/handlers/validation/builtin/v12
- stop using mock cc as it adds more complexity than needed when you can simply "create" lscc records

#### Additional details

End goal is to remove common/mocks entirely

#### Related issues

[FAB-16437](https://jira.hyperledger.org/browse/FAB-16437)

### CODE REVIEW

**Review by jyellick — APPROVED** [review]
You are a saint for wading through this test code to remove these deps, a significant improvement.

**Review by ale-linux — APPROVED** [review]

**Comment by ale-linux** [comment]
THANKS! We probably want to do the same for the other validator forks

**Comment by mastersingh24** [comment]
> THANKS! We probably want to do the same for the other validator forks

yep

File: core/handlers/validation/builtin/v12/validation_logic.go [hunk 1]
```diff
@@ -711,7 +711,7 @@ func (vscc *Validator) ValidateLSCCInvocation(
 			/*****************************************************/
 			pol := cdLedger.InstantiationPolicy
 			if pol == nil {
-				return policyErr(fmt.Errorf("No instantiation policy was specified"))
+				return policyErr(fmt.Errorf("-No instantiation policy was specified"))
```
[the remark this answers is not in the document]
Comment (mastersingh24) [reply core/handlers/validation/builtin/v12/validation_logic.go]: I'll fix this when I do v13

[the remark this answers is not in the document]
Comment (mastersingh24) [reply core/handlers/validation/builtin/v12/validation_logic.go]: Oops -- I did fix it  :-)

**Pull request merged**